### PR TITLE
Backup existing parameters.XXX files with date before writing

### LINKED ISSUE #391: parameters.txt.old ->parameters.txt.old_ss.MM.hh.dd.mm.yyyy or similar
Add timestamp to parameters.txt.old (eg parameters.txt.old.ss.mm.hh.dd.mm.yyy or something similar) when renamed so that it becomes unique and not lost the second time it is overwritten.

Requested from user at ERT customer voice meeting.

### PR DESCRIPTION
**Pull request by andreabrambilla**
**Issue**
Resolves https://github.com/equinor/ert/issues/391

### CODE REVIEW

**Review by ManInFez — APPROVED** [review]
Ship it!!

File: lib/enkf/tests/value_export.cpp [hunk 1]
```diff
@@ -19,16 +19,65 @@
...
 
 
+namespace {
+
+// This is super ugly but I could not find any way to do this without using a
+// a global variable. Of course the correct thing to do would be to implement
+// a new walk_directory that either collects first and return (include
+// max_depth, remove callbacks), or behaves like an iterator. But that would
+// require quite some efforts, and it's probably not worth it for just one
+// test. Especially considering that C++17 includes the filesystem library that
```
Comment (markusdregi) [inline lib/enkf/tests/value_export.cpp:43]: Considering that `libres` is now moving towards having a Python-only API; I would say that we can use whatever compiler floats our boat ;)

File: lib/enkf/tests/value_export.cpp [hunk 2]
```diff
@@ -132,6 +203,28 @@ void test_export_txt() {
...
+      std::begin(tree),
+      std::end(tree),
+      [](std::string const& l, std::string const& r) -> bool {
+          return l.size() < r.size();
+      });
+
+  test_assert_string_equal(tree[0].c_str(), "path/parameters.txt");
+  test_assert_string_equal(
+      tree[2].substr(0, 29).c_str(),
+      "path/parameters.txt_backup_20"); // Fix this in 80 years
```
Comment (markusdregi) [inline lib/enkf/tests/value_export.cpp:224]: Already looking forward to it 😎

Would it be possible to have a slightly more exact test of this?

Comment (andreabrambilla) [reply lib/enkf/tests/value_export.cpp:224]: given how painful it is to deal with time and date in c++, I'd rather wait until we move everything to python (which hopefully will happen before this test start failing :) )

**Pull request merged**